x-pack/libbeat/common/aws/acker.go

```diff
@@ -18,54 +18,40 @@ import (
 // more S3 objects.
 type EventACKTracker struct {
 	sync.Mutex
-	PendingACKs int64
-	ctx         context.Context
-	cancel      context.CancelFunc
+	PendingACKs *atomic.Uint64
```

Comment:
Switching from mutex to atomic, keeping this new `EventACKTracker` implementation on the old codebase, reduced 2.5/4x overall benchmark suite time for 2 load type out of 3